### ISSUE TITLE
Explode on lazy frame seems to misplace column.

### ISSUE THREAD

**Issue by bmschmidt** [pola-rs/polars#2657]
Thanks for your work.

#### Are you using Python or Rust?

Python

#### Which feature gates did you use?

This can be ignored by Python users.

#### What version of polars are you using?

0.13.1

#### What operating system are you using polars on?

OS X

#### Describe your bug.

When exploding a column in a lazy data frame, the column seems to somehow get lost among the expressions throwing a RuntimeError.
 
Depending how many steps happen after the explode, the error message changes. (In my initial chain,  got "This operation is likely not what you want. Please open an issue if you really want to do this.") 

This bug is a regression from earlier behavior; I had this code running in November.

#### What are the steps to reproduce the behavior?

```
import polars as pl

pl.DataFrame({'text': ["One two three four", "uno dos tres quatro"]})\
   .lazy()\
   .select([pl.col('text').str.split(" ").alias("tokens")])\
   .with_row_count()\
   .explode(pl.col("tokens"))\
   .select(["row_nr", "tokens"])\
   .collect()
```

#### What is the actual behavior?

Throws a runtime error:

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/var/folders/32/kljptc3d7_vgm80m4l25vh6c0000gn/T/ipykernel_10630/1759863222.py in <module>
      7    .explode(pl.col("tokens"))\
      8    .select(["row_nr", "tokens"])\
----> 9    .collect()

/opt/homebrew/Caskroom/miniforge/base/envs/analytics/lib/python3.10/site-packages/polars/internals/lazy_frame.py in collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, string_cache, no_optimization, slice_pushdown)
    396             slice_pushdown,
    397         )
--> 398         return pli.wrap_df(ldf.collect())
    399 
    400     def fetch(

RuntimeError: Any(NotFound("tokens"))
```
#### What is the expected behavior?

Same as happens without lazy:

```py
import polars as pl

pl.DataFrame({'text': ["One two three four", "uno dos tres quatro"]})\
   .select([pl.col('text').str.split(" ").alias("tokens")])\
   .with_row_count()\
   .explode(pl.col("tokens"))\
   .select(["row_nr", "tokens"])
```

**Comment by ritchie46** [comment]
Thanks. I have got a fix in #2658

**Comment by bmschmidt** [comment]
Wow that was fast! Thanks!
